### PR TITLE
[DDW-1180] Resync wallet

### PR DESCRIPTION
**Pull request by nikolaglumac**
This PR implements the "Resync wallet" feature which enables you to resync individual wallets in Daedalus UI.

## Todos

- [x] Implement UI
- [x] Inject translations
- [x] Implement API and data logic
- [x] Connect UI with data logic
- [x] Implement acceptance tests

## Screenshots

Screenshot 2020-01-20 at 10 36 01
Screenshot 2020-01-20 at 10 36 10
Screenshot 2020-01-20 at 10 36 19
Screenshot 2020-01-20 at 10 36 23

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Try to click on "Resync Wallet" button on the "Wallet Settings" screen for both Rewards and Balance wallet.

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by daniloprates** [comment]
Screen Shot 2020-01-17 at 14 18 27

I'm getting these errors, sometimes

**Comment by DmitriiGaico** [comment]
Same for me. Trying to understand how to trigger them.
image

**Comment by daniloprates** [comment]
error

**Comment by daniloprates** [comment]
The "Resync wallet" button loading state is not connected to the syncing state, which makes it look strange and allows the user to click the button multiple times, even before the syncing starts to happen.

wallet

**Comment by DmitriiGaico** [comment]
> The "Resync wallet" button loading state is not connected to the syncing state, which makes it look strange and allows the user to click the button multiple times, even before the syncing starts to happen.
> 
> wallet

Yes, also noticed that sometimes nothing happens after clicking the button. Sometimes there is kinda of delay between Click and Process started.

n you click and process starts.

File: source/renderer/app/components/wallet/settings/WalletSettings.js [hunk 1]
```diff
@@ -172,8 +188,23 @@ export default class WalletSettings extends Component<Props, State> {
         styles.deleteWalletBox,
         styles.legacyWallet,
       ]);
+      const resyncWalletBoxStyles = classNames([
+        styles.resyncWalletBox,
+        styles.legacyWallet,
+      ]);
       return (
         <div className={styles.component}>
+          <BorderedBox className={resyncWalletBoxStyles}>
```
Comment (daniloprates) [inline source/renderer/app/components/wallet/settings/WalletSettings.js]: What about creating dedicated components for Sync and Delete?

Comment (nikolaglumac) [reply source/renderer/app/components/wallet/settings/WalletSettings.js]: @daniloprates please add those if you have time today.

**Comment by nikolaglumac** [comment]
@daniloprates @DmitriiGaico please test starting from clean state directory.

We can not know when the actual resync will start. We can only tell when our initial resync trigger request finishes. Perhaps we should add some type of success overlay which explains what happens next?

File: source/renderer/app/components/wallet/settings/WalletSettings.js (hunk 1 above)
Comment (nikolaglumac) [reply source/renderer/app/components/wallet/settings/WalletSettings.js]: @daniloprates please keep in mind that we have a bit different layout for Balance vs Rewards wallets!

**Comment by nikolaglumac** [comment]
@daniloprates the case where Node crashes is a backend issue and it is being investigated...

File: source/renderer/app/components/wallet/settings/WalletSettings.js (hunk 1 above)
Comment (nikolaglumac) [reply source/renderer/app/components/wallet/settings/WalletSettings.js]: @daniloprates done: https://github.com/input-output-hk/daedalus/pull/1822/commits/8b9cfab1ce1f253e31c5252d6aab4413dc2481f6

**Comment by nikolaglumac** [comment]
@daniloprates @DmitriiGaico all issues are fixed now - I have added 1-second throttling for the resync wallet request and made sure the button stays in "spinning" state until the restoration notification is eventually shown.

There should be no JS errors thrown any more...

**Comment by a-rukin** [comment]
image
for the Incentivized testnet theme only we have extra padding on top (storybook)

**Comment by a-rukin** [comment]
image
plus we have some additional padding around this block 🤔

**Comment by a-rukin** [comment]
image
not sure if this wasn't fixed somewhere else, but here it's not centered vertically 🤷‍♂

**Comment by a-rukin** [comment]
image
spinner is too big here 
@nikolaglumac I couldn't find how to tweak this in inspector to tell you proper size (( but we can do it on a call

**Comment by nikolaglumac** [comment]
@a-rukin all styling issues have now been fixed!

**Pull request merged**